src/routes/api-routes.js

```diff
@@ -0,0 +1,219 @@
+      ]
+  }
+]
+
+// Data access functions
+// TODO: transfer these functions to a separate file
+
+// function length
+// description: Returns length property of a given stream
+// inputs: stream_id:Int
+// returns: Int
+async function length (stream_id) {
+  var len = null
+  if (stream_id < dummy.length && stream_id >= 0) {
+    len = dummy[stream_id].length
+  }
+  return len
+}
+
+// function load
+// description: Returns an interval of events in a given stream
+// inputs: stream_id:Int, from:Int, to:Int
+// returns: [Object]
+async function load (stream_id, from, to) {
+  var result = []
+  if (stream_id < dummy.length && stream_id >= 0) {
+    var events = dummy[stream_id].events
+    if (from >= 0 && to >= from && from < events.length && to < events.length) {
+      result = events.slice(from, to+1)
+    }
+  }
+  return result
+}
+
+// function push
+// description: Include an event in a given stream
+// inputs: stream_id:Int, data:String
+// returns: Object -- success/failure indicator
+async function push (stream_id, data) {
+  var success = false
+  if (stream_id < dummy.length && stream_id >= 0) {
+    var len = dummy[stream_id].length
+    dummy[stream_id].events[len] = { index: len, data: data }
+    dummy[stream_id].length += 1
+    success = true
+  }
+  return success
+}
+
+// =============== Route URLs =============== //
+const length_url = api_base_url + '/streams/:stream_id'
+const load_url = api_base_url + '/streams/:stream_id/events/:from/:to'
+const push_url = api_base_url + '/streams/:stream_id/events'
+
+// =============== Route Schemas =============== //
+const length_schema =
+{
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        stream: { type: 'string' },
+        length: { type: 'integer' },
+        msg:  { type: 'string' },
+      },
+    }
+  }
+}
+
+const load_schema =
+{
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        stream: { type: 'string' },
+        events: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              index: {type: 'integer'},
+              data: {type: 'string'},
+            },
+          },
+        },
+      },
+    },
+  },
+}
+
+const push_schema =
+{
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        success: {type: 'boolean'},
+        msg: {type: 'string'},
+      },
+    },
+    201: {
+      type: 'object',
+      properties: {
+        success: {type: 'boolean'},
+        msg: {type: 'string'},
+      },
+    },
+  },
+}
+
+// =============== Route Handlers =============== //
+async function length_handler (req, res) {
+  const stream_id = req.params.stream_id
+  const len = await length(stream_id)
+  const error_msg = (len === null) ? 'stream_id not found' : '';
+
+  const resp =
+  {
+    stream: stream_id,
+    length: len,
+    msg : error_msg,
+  }
+
+  return JSON.stringify(resp)
```

Comment:
You are just returning a string. The content type is responding with `plain/text` while it should be responding with `application/json`. That is probably something related to how you configured fastify.